src/pages/query-view/query-view.js

```diff
@@ -55,8 +55,8 @@ const QueryView = props => {
         cancelQuery={cancelQuery}
         setQueryText={setQueryText}
       />
-      <div id="dataset-preview-table">
-        <div id="numRecords">{numRecords}</div>
+      <div data-testid="dataset-preview-table" id="dataset-preview-table">
+        <div data-testid="numRecords" id="numRecords">{numRecords}</div>
```

Comment:
I don't think these provide any more useful selection than simply using id

Reply:
Adding ```data-testid``` attribute to the element gives us a targeted selector that's only used for testing.

The ```data-testid``` attribute will not change from CSS style or JS behavioral changes, meaning it’s not coupled to the behavior or styling of an element.

Additionally, it makes it clear to everyone that this element is used directly by test code.